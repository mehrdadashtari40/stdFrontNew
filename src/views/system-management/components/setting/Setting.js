import React, {useContext} from 'react';
import getAuthonticatedJSON from '../../../../common/utils/functions/getAuthenticatedJSON';
import postAuthenticatedJSON from '../../../../common/utils/functions/postAuthenticatedJSON';
import {WidgetGrid, JarvisWidget, smallBox} from '../../../../common';
import $ from 'jquery';
import CardContent from "@mui/material/CardContent";
import MaterialDataTable from "../../../../common/tables/components/MaterialDataTable";
import Card from "@mui/material/Card";
import {compose} from "redux";
import {withTranslation} from "react-i18next";
import {connect} from "react-redux";
import * as Actions from "../../../process-management/ProcessManagementActions";
import {AppConfig} from "../../../../appConfig";

class Setting extends React.Component {
    static contextType = AppConfig;
    constructor(props) {
        super(props);
        this.state = {
            value: "",
            dashboards:[]
        }
    }

    componentDidMount() {
        const config = this.context;
        let self = this;
        getAuthonticatedJSON(config.apiServer + 'anrsetting/default_route').then((res) => {
            self.setState({value:res});
            $("input[value=" + res + "]").attr('checked', 'checked');
        });
        getAuthonticatedJSON(config.apiServer + 'biarian/my-dashboard-report-list').then((res) => {
            self.setState({dashboards:res});
            $("input[value=" + self.state.value + "]").attr('checked', 'checked');
        });
    }

    saveDefault = () => {
        const config = this.context;
        let self = this;
        let URL = config.apiServer + "anrsetting";
        let request = {
            item_key: 'default_route',
            item_value: $("input[type='radio'][name='defaultPage']:checked").val()
        };
        postAuthenticatedJSON(URL, request)
            .then(res => {
                smallBox({
                    title: self.props.t('Successfully'),
                    content: '<i>' + self.props.t("Data has been saved successfully") + '</i>',
                    color: '#659265',
                    iconSmall: 'fa fa-check fa-2x fadeInRight animated',
                    timeout: 4000,
                });
            });
    }
    radioChange = e => {
        this.setState({
            value: e.target.value
        })
    }

    render() {
        const config = this.context;

        return (<>
            <Card id="content">
                <CardContent>
                    <h5 className={'table-header-title'}>
                        <i className={"fal fa-user table-header-icon"}></i>
                        تنظیمات نمایه
                    </h5>

                    <div className="row">
                        <div className="col-sm-12 col-md-12 col-lg-12" style={{paddingBottom:20 , marginLeft:20}}>
                            <label className="radio" style={{marginRight: 20}}>
                                {/*<input type="radio" name="defaultPage" id="default" value="#/default" />*/}
                                <input type="radio" name="defaultPage" id="default" value="inbox"
                                       onChange={this.radioChange}/>
                                کارتابل</label>
                            <label className="radio" style={{marginRight: 20}}>
                                <input type="radio" name="defaultPage" id="analytical" value="DashboardBi"
                                       onChange={this.radioChange}/>
                                داشبورد تحلیلی</label>

                            {this.state.dashboards.map((x,key)=>{
                                return (<label key={key} className="radio" style={{marginRight: 20}}>
                                    <input type="radio" name="defaultPage" value={x.route}
                                           onChange={this.radioChange}/>
                                    {x.title}</label>)
                            })}
                            {/*<label className="radio" style={{marginRight: 20}}>*/}
                            {/*    <input type="radio" name="defaultPage" id="guest" value="dashboard/guest"*/}
                            {/*           onChange={this.radioChange}/>*/}
                            {/*    داشبورد مهمان</label>*/}
                            <button type="button" className="btn btn-success"
                                    onClick={() => this.saveDefault()}>ثبت
                            </button>
                        </div>
                    </div>
                </CardContent>
            </Card>
        </>)
    }

}
export default withTranslation()(Setting);