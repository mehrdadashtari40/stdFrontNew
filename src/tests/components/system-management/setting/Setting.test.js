import React from 'react';
import { shallow } from 'enzyme';
import Setting from '../../../../views/system-management/components/setting/Setting';

// const store = mockStore();

jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));

test('Make Snapshot file , Setting Component', () => {
	const wrapper = shallow(<Setting />);
	expect(wrapper).toMatchSnapshot();
});

it('invokes `componentDidMount` when mounted', () => {
	jest.spyOn(Setting.prototype, 'componentDidMount');
	shallow(<Setting />);
	expect(Setting.prototype.componentDidMount).toHaveBeenCalled();
	Setting.prototype.componentDidMount.mockRestore();
});

it('radioChange HaveBeenCalled when radio buttonChanged?', () => {
	const wrapper = shallow(<Setting />);

	const instance = wrapper.instance();
	const spy = jest.spyOn(instance, 'radioChange');

	instance.forceUpdate();

	const input = wrapper.find('input').at(0);
	input.simulate('change', { target: { value: 'test' } });
	expect(spy).toHaveBeenCalled();
});
it('saveDefault HaveBeenCalled ?', () => {
	const wrapper = shallow(<Setting />);

	const instance = wrapper.instance();
	const spy = jest.spyOn(instance, 'saveDefault');

	instance.forceUpdate();

	const Button = wrapper.find('button');
	Button.simulate('click');
	expect(spy).toHaveBeenCalled();
});



test('saveDefault Function  ', () => {
	const wrapper = shallow(<Setting />);

	const instance = wrapper.instance();
	expect(instance.saveDefault).toBeDefined();
});
