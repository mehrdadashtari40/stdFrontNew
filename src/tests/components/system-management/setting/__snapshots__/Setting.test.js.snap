// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , Setting Component 1`] = `
<WidgetGrid>
  <div
    id="content"
  >
    <div
      className="inbox-nav-bar no-content-padding"
    >
      <div
        className="big-breadcrumbs"
      >
        <h1
          className="page-title txt-color-blueDark"
        >
          <i
            className="fa fa-fw fa-inbox"
          />
          تنظیمات
          <span>
            <span
              className="page-title-separator"
            >
              <i
                className="fa fa-angle-right"
              />
                
            </span>
            صفحه نخست
          </span>
        </h1>
      </div>
      <div
        className="row"
      >
        <article
          className="col-sm-12 col-md-12 col-lg-12"
        >
          <div
            className="inbox-checkbox-triggered"
          >
            <JarvisWidget
              collapsed={false}
              color={false}
              colorbutton={false}
              custombutton={false}
              deletebutton={false}
              editbutton={false}
              fullscreenbutton={true}
              hidden={false}
              load={false}
              refresh={false}
              sortable={true}
              togglebutton={false}
            >
              <header>
                <nav
                  className="inbox_header_item_bx"
                >
                  <p>
                    انتخاب صفحه نخست
                  </p>
                </nav>
              </header>
              <div
                className="inbox-body1 no-content-padding1 no-padding"
                id="inbox-content"
              >
                <fieldset>
                  <section
                    style={
                      Object {
                        "padding": 50,
                      }
                    }
                  >
                    <label
                      className="label"
                    >
                      گزینه های انتخابی
                    </label>
                    <div
                      className="inline-group"
                    >
                      <label
                        className="radio"
                        style={
                          Object {
                            "marginRight": 20,
                          }
                        }
                      >
                        <input
                          id="default"
                          name="defaultPage"
                          onChange={[Function]}
                          type="radio"
                          value="default"
                        />
                        <i />
                        کارتابل
                      </label>
                      <label
                        className="radio"
                        style={
                          Object {
                            "marginRight": 20,
                          }
                        }
                      >
                        <input
                          id="inbox"
                          name="defaultPage"
                          onChange={[Function]}
                          type="radio"
                          value="outlook/folder/todo"
                        />
                        <i />
                        داشبورد
                      </label>
                      <label
                        className="radio"
                        style={
                          Object {
                            "marginRight": 20,
                          }
                        }
                      >
                        <input
                          id="analytical"
                          name="defaultPage"
                          onChange={[Function]}
                          type="radio"
                          value="dashboard/analytical"
                        />
                        <i />
                        داشبورد تحلیلی
                      </label>
                      <label
                        className="radio"
                        style={
                          Object {
                            "marginRight": 20,
                          }
                        }
                      >
                        <input
                          id="guest"
                          name="defaultPage"
                          onChange={[Function]}
                          type="radio"
                          value="dashboard/guest"
                        />
                        <i />
                        داشبورد مهمان
                      </label>
                    </div>
                    <br />
                    <div
                      className="row"
                    >
                      <button
                        className="btn btn-success"
                        onClick={[Function]}
                        type="button"
                      >
                        ثبت
                      </button>
                    </div>
                  </section>
                </fieldset>
              </div>
              <a
                className="btn btn-primary pull-right hidden-desktop visible-tablet"
                id="compose-mail-mini"
              >
                <strong>
                  <i
                    className="fa fa-file fa-lg"
                  />
                </strong>
              </a>
            </JarvisWidget>
          </div>
        </article>
      </div>
    </div>
  </div>
</WidgetGrid>
`;
